core/iwasm/common/gc/gc_object.c

```diff
@@ -750,3 +791,296 @@ wasm_obj_unset_gc_finalizer(wasm_exec_env_t exec_env, void *obj)
+        return NULL;
+    }
+
+#if WASM_ENABLE_INTERP != 0
+    if (module_inst->module_type == Wasm_Module_Bytecode) {
+        WASMModule *module = ((WASMModuleInstance *)module_inst)->module;
+        rtt_type = wasm_stringref_rtt_type_new(WASM_TYPE_STRINGREF,
+                                               module->stringref_rtts,
+                                               &module->rtt_type_lock);
+    }
+#endif
+#if WASM_ENABLE_AOT != 0
+    if (module_inst->module_type == Wasm_Module_AoT) {
+        AOTModule *module =
+            (AOTModule *)((AOTModuleInstance *)module_inst)->module;
+        /* TODO: add stringref_rtts in AOTModule */
+        rtt_type = wasm_stringref_rtt_type_new(
+            WASM_TYPE_STRINGREF, module->rtt_types, &module->rtt_type_lock);
+    }
+#endif
+
+    if (!rtt_type) {
+        return NULL;
+    }
+
+    stringref_obj->header = (WASMObjectHeader)rtt_type;
+    stringref_obj->str_obj = str_obj;
+
+    wasm_obj_set_gc_finalizer(
+        exec_env, (wasm_obj_t)stringref_obj,
+        (wasm_obj_finalizer_t)wasm_stringref_obj_finalizer, NULL);
+
+    return stringref_obj;
+}
+
+WASMStringviewWTF8ObjectRef
+wasm_stringview_wtf8_obj_new(WASMExecEnv *exec_env, const void *str_obj)
+{
+    void *heap_handle = get_gc_heap_handle(exec_env);
+    WASMStringviewWTF8ObjectRef stringview_wtf8_obj;
+    WASMModuleInstanceCommon *module_inst =
+        wasm_runtime_get_module_inst(exec_env);
+    WASMRttTypeRef rtt_type = NULL;
+
+    if (!(stringview_wtf8_obj =
+              gc_obj_malloc(heap_handle, sizeof(WASMStringviewWTF8Object)))) {
+        return NULL;
+    }
+
+#if WASM_ENABLE_INTERP != 0
+    if (module_inst->module_type == Wasm_Module_Bytecode) {
+        WASMModule *module = ((WASMModuleInstance *)module_inst)->module;
+        rtt_type = wasm_stringref_rtt_type_new(WASM_TYPE_STRINGVIEWWTF8,
+                                               module->stringref_rtts,
+                                               &module->rtt_type_lock);
+    }
+#endif
+#if WASM_ENABLE_AOT != 0
+    if (module_inst->module_type == Wasm_Module_AoT) {
+        AOTModule *module =
+            (AOTModule *)((AOTModuleInstance *)module_inst)->module;
+        /* TODO: add stringref_rtts in AOTModule */
+        rtt_type = wasm_stringref_rtt_type_new(WASM_TYPE_STRINGVIEWWTF8,
+                                               module->rtt_types,
+                                               &module->rtt_type_lock);
+    }
+#endif
+
+    if (!rtt_type) {
+        return NULL;
+    }
+
+    stringview_wtf8_obj->header = (WASMObjectHeader)rtt_type;
+    stringview_wtf8_obj->str_obj = str_obj;
+
+    wasm_obj_set_gc_finalizer(
+        exec_env, (wasm_obj_t)stringview_wtf8_obj,
+        (wasm_obj_finalizer_t)wasm_stringview_wtf8_obj_finalizer, NULL);
+
+    return stringview_wtf8_obj;
+}
+
+WASMStringviewWTF16ObjectRef
+wasm_stringview_wtf16_obj_new(WASMExecEnv *exec_env, const void *str_obj)
```

Comment:
The implementation of the four APIs are very similar, maybe we can unify them in an internal function, like:
```C
static WASMObjectRef
stringref_obj_new(WASMExecEnv *exec_env, uint32 type, const void *str_obj,
                             int32 pos)
```
And these APIs call it with individual types?

Reply:
Great idea, I will add it.